### PR TITLE
CalorimeterHitDigi: use EvaluatorSvc for corrMeanScale

### PR DESCRIPTION
**Pull request by veprbl**
This is a sequel to #1349, now for digitization.

### CODE REVIEW

File: src/algorithms/calorimetry/CalorimeterHitDigi.cc [hunk 1]
```diff
@@ -96,11 +98,25 @@ void CalorimeterHitDigi::init() {
...
     if (!m_cfg.fields.empty()) {
         for (auto & field : m_cfg.fields) {
-            id_inverse_mask |= id_desc.field(field)->mask();
+            id_inverse_mask |= id_spec.field(field)->mask();
         }
         debug("ID mask in {:s}: {:#064b}", m_cfg.readout, id_mask);
     }
     id_mask = ~id_inverse_mask;
+
+    std::function hit_to_map = [this](const edm4hep::SimCalorimeterHit &h) {
```
Comment (wdconinc) [inline src/algorithms/calorimetry/CalorimeterHitDigi.cc:107]: Optional, deferrable. Can this be reused from CalorimeterHitReco, somehow? It's essentially the same code. Templated lambda with concept `has_cellID`?

**Review by wdconinc — APPROVED** [review]
Not sure I like the computational complexity this adds for every hit, when at this point (and probably for some detectors never) this isn't used. For a corrMeanScale of 1, this changes a noop to a looped set of string lookups in the index, right?

File: src/algorithms/calorimetry/CalorimeterHitDigi.cc (hunk 1 above)
Comment (veprbl) [reply src/algorithms/calorimetry/CalorimeterHitDigi.cc:107]: That is understood. What goes into the function is still in flux at this point, will likely add more fields or change how this works anyway.

**Review by ruse-traveler — APPROVED** [review]
Nice! 🙌

**Pull request merged**